utxorpc/query.go

```diff
@@ -36,9 +41,32 @@ func (s *queryServiceServer) ReadParams(
 ) (*connect.Response[query.ReadParamsResponse], error) {
 	fieldMask := req.Msg.GetFieldMask()
 
-	log.Printf("Got a ReadParams request with fieldMask %v", fieldMask)
+	s.utxorpc.config.Logger.Info(
+		fmt.Sprintf(
+			"Got a ReadParams request with fieldMask %v",
+			fieldMask,
+		),
+	)
 	resp := &query.ReadParamsResponse{}
 
+	lsParams := s.utxorpc.config.LedgerState.GetCurrentPParams()
+	// TODO: support all eras for protocol parameters
+	protoParams := lsParams.(lconway.ConwayProtocolParameters)
```

Comment:
You can remove this type cast if you change the return type on `GetCurrentPParams()` above